src/app/modules/item/components/item-children-edit/item-children-edit.component.ts

```diff
@@ -50,4 +52,11 @@ export class ItemChildrenEditComponent implements OnChanges {
     }
   }
 
+  orderChanged(): void {
+    if (this.state == 'error') return;
+    this.parentForm?.patchValue({
+      children: this.data.map((child, idx) => ({ item_id: child.id, order: idx })),
+    });
```

Comment:
I do not have the impression that's the way to go to update your data to your parent form. Is it?

https://ritchiejacobs.be/angular-custom-form-component
(I haven't checked it in details yet)

Reply:
The reason why I'm using that this.data variable is because of the p-table. 

I'm not the one changing the order of this.data, but primeng. And the p-table can't change the value inside a reactive form. So it seems like I have to use a variable and copy the content of that variable into the form.

Wouldn't creating a custom form component just move the problem?

Reply:
What I meant is I am not sure you are supposed to change `this.parentForm?.patchValue` from here. 
Either it is not a form component and the main form fetches the info from this component, or it is a form component and it behaves as a form component. Probably option (1) is easier and more relevant here, but if so, there is no `parentForm` to be known by this component.

Reply:
Done